src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,167 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exeptions.UserRegistrationException;
+import core.basesyntax.model.User;
+import java.time.LocalDateTime;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "test_login_created_at " + LocalDateTime.now();
+    private static final String VALID_PASSWORD = "Mine!$VAl1DP@ZSW0Rd!)(*&^%$#@";
+    private static final int VALID_AGE = 20;
+    private static final String VALID_EDGE_LOGIN = "testLo";
+    private static final String VALID_EDGE_PASSWORD = "MiP@33";
+    private static final int VALID_EDGE_AGE = 18;
+    private static final String INVALID_SHORT_LOGIN = "testL";
+    private static final String INVALID_SHORT_PASSWORD = "MiP@3";
+    private static final int INVALID_AGE = 17;
+    private static final String INVALID_EMPTY_LOGIN = "";
+    private static final String INVALID_EMPTY_PASSWORD = "";
+    private static final int INVALID_ZERO_AGE = 0;
+
+    private RegistrationServiceImpl registrationService;
+    private User user;
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }

```

Comment:
i guess we should create new User here .
and registrationService = new RegistrationServiceImpl(); don't need to create before each

Reply:
The idea of User creation - each time I create user with new fields according to test goal so no need to create before each since I would need to setup its field anyway

I have class field
private RegistrationServiceImpl registrationService; - and I create new object for each test